test/functional/wallet_importdescriptors.py

```diff
@@ -687,11 +690,26 @@ def run_test(self):
         descriptor["timestamp"] = 0
         descriptor["next_index"] = 0
 
-        batch = []
-        batch.append(encrypted_wallet.walletpassphrase.get_request("passphrase", 3))
-        batch.append(encrypted_wallet.importdescriptors.get_request([descriptor]))
+        encrypted_wallet.walletpassphrase("passphrase", 99999)
+        t = threading.Thread(target=encrypted_wallet.importdescriptors, args=([descriptor],))
+
+        with self.nodes[0].assert_debug_log(expected_msgs=[f'Rescan started from block 0f9188f13cb7b2c71f2a335e3a4fc328bf5beb436012afca590b1a11466e2206... (slow variant inspecting all blocks)'], timeout=5):
+            t.start()
+
+        # Set the passphrase timeout to 1 to test that the wallet remains unlocked during the rescan
+        self.nodes[0].cli("-rpcwallet=encrypted_wallet").walletpassphrase("passphrase", 1)
+
+        try:
+            self.nodes[0].cli("-rpcwallet=encrypted_wallet").walletlock()
```

Comment:
It seems (?) we want this test to fail if `walletlock` succeeds, but that's not currently the case - is that on purpose? Apologies if I'm misunderstanding, I didn't dive super deep yet so feel free to keep your answer brief.